fire_atlas_editor/lib/screens/editor_screen/widgets/selection_canvas/canvas_board.dart

```diff
@@ -64,37 +70,42 @@ class CanvasBoardState extends State<CanvasBoard> {
   }
 
   void _handleMove(DragUpdateDetails details) {
-    setState(() {
-      final x = details.localPosition.dx - _lastDrag.dx;
-      final y = details.localPosition.dy - _lastDrag.dy;
-
-      if (_currentTool == CanvasTools.MOVE) {
-        _translateX += x;
-        _translateY += y;
-      } else if (_currentTool == CanvasTools.SELECTION) {
-        _finishSelection(_calculateIndexClick(details.localPosition));
-      }
-
-      _lastDrag = details.localPosition;
-    });
+    if (_lastDrag != null) {
+      final _last = _lastDrag!;
+      setState(() {
+        final x = details.localPosition.dx - _last.dx;
+        final y = details.localPosition.dy - _last.dy;
+
+        if (_currentTool == CanvasTools.MOVE) {
+          _translateX += x;
+          _translateY += y;
+        } else if (_currentTool == CanvasTools.SELECTION) {
+          _finishSelection(_calculateIndexClick(details.localPosition));
+        }
+
+        _lastDrag = details.localPosition;
+      });
+    }
   }
 
   void _handleMoveEnd() {
-    setState(() {
-      if (_currentTool == CanvasTools.SELECTION) {
-        _finishSelection(_calculateIndexClick(_lastDrag));
-      }
-      _lastDrag = _dragStart = null;
-    });
+    if (_lastDrag != null) {
```

Comment:
Just a pet peeve, but I personally would write this as:
```dart
if (_lastDrag == null) {
  return;
}

// Your code
```

It saves on indenting and it makes sure that all the "bad" routes are handled before you go with the happy flow.

Reply:
Changed

Reply:
Uh it used `!=` now, not sure if that was intended?

Reply:
wow, this is wrong! Weird enough apparently nothing broke because of this, I will fix either way